### PR TITLE
Add initial_value column and constraints for rw consistency and initial values

### PR DESCRIPTION
**Pull request by z2trillion**

### CODE REVIEW

File: zkevm-circuits/src/state_circuit/constraint_builder.rs [hunk 1]
```diff
@@ -306,6 +352,10 @@ impl<F: Field> Queries<F> {
         self.first_access.clone()
     }
 
+    fn not_first_access(&self) -> Expression<F> {
+        self.first_access.clone()
```
Comment (lispc) [inline zkevm-circuits/src/state_circuit/constraint_builder.rs:356]: typo?

Comment (z2trillion) [reply zkevm-circuits/src/state_circuit/constraint_builder.rs:356]: yup

File: zkevm-circuits/src/state_circuit.rs [hunk 2]
```diff
@@ -44,6 +44,8 @@ pub struct StateConfig {
     sort_keys: SortKeysConfig,
     is_write: Column<Advice>,
     value: Column<Advice>,
+    prev_value: Column<Advice>,
+    committed_value: Column<Advice>,
```
Comment (miha-stopar) [inline zkevm-circuits/src/state_circuit.rs:48]: Would perhaps add a short description what is `committed_value` and that it serves for reducing the calls into MPT.

File: zkevm-circuits/src/state_circuit/lexicographic_ordering.rs [hunk 3]
```diff
@@ -233,7 +237,10 @@ impl Config {
             || Ok(limb_difference.invert().unwrap()),
         )?;
 
-        Ok(())
+        Ok(!matches!(
+            index,
+            LimbIndex::RwCounter0 | LimbIndex::RwCounter1
```
Comment (miha-stopar) [inline zkevm-circuits/src/state_circuit/lexicographic_ordering.rs:242]: Perhaps some comment that it returns `is_first_access`.

File: zkevm-circuits/src/state_circuit/constraint_builder.rs [hunk 4]
```diff
@@ -28,9 +28,14 @@ pub struct Queries<F: Field> {
     pub field_tag: Expression<F>,
     pub storage_key: RlcQueries<F, N_BYTES_WORD>,
     pub value: Expression<F>,
+    pub prev_value: Expression<F>,
+    pub value_prev: Expression<F>,
```
Comment (miha-stopar) [inline zkevm-circuits/src/state_circuit/constraint_builder.rs:32]: I remember I was trying to come out with some other names for `prev_value` and `value_prev` in the first version of the state circuit, but didn't have some good idea - but perhaps just add some comment what's the difference between the two?

File: zkevm-circuits/src/state_circuit/constraint_builder.rs [hunk 5]
```diff
@@ -101,6 +106,21 @@ impl<F: Field> ConstraintBuilder<F> {
...
+            cb.require_zero(
+                "non-first access reads don't change value",
+                q.is_read() * (q.value.clone() - q.prev_value.clone()),
+            );
+            cb.require_zero(
+                "prev_value matches previous value",
+                q.value_prev.clone() - q.prev_value.clone(),
+            );
+            cb.require_zero(
+                "committed value doesn't change in a group",
```
Comment (miha-stopar) [inline zkevm-circuits/src/state_circuit/constraint_builder.rs:120]: Perhaps a short desc that group means changes at the same key?

File: zkevm-circuits/src/state_circuit/constraint_builder.rs (hunk 4 above)
Comment (z2trillion) [reply zkevm-circuits/src/state_circuit/constraint_builder.rs:31]: This naming conundrum was the main reason I ended up getting rid of the prev_value column :)

**Review by lispc — APPROVED** [review]
lgtm.    We may revisit the design of RwTableTag::CallContext | RwTableTag::TxReceipt later

**Review by ed255 — COMMENTED** [review]
All looks good!
I've added some notes about adding comments.  Please also take a look at Miha's suggestions to add comments :)

**Review by ed255 — APPROVED** [review]
LGTM! Thanks for adding the comments :)

**Pull request merged**